### ISSUE TITLE
Missing space before `link` in `formattedref`

### ISSUE THREAD

**Issue by Intelligent2013** [metanorma/metanorma-ieee#161]
Source issue: https://github.com/metanorma/metanorma-ieee/issues/160

The hyperlink renders without indent due missed space in the presentation XML:
![image](https://user-images.githubusercontent.com/5559926/198985846-81f71b87-1818-4d2d-80ed-299cc421af1d.png)

```xml
<bibitem id="mozilla">
<formattedref format="application/x-isodoc+xml">Mozilla, Fundamentals of Web XR<link target="https://developer.mozilla.org/en-US/docs/Web/API/WebXR_Device_API/Fundamentals"/> [Mozilla Web XR] </formattedref>
<title>Fundamentals of Web XR</title>
```

**Comment by opoudjis** [comment]
Original:

````
 [[[mozilla,Mozilla Web XR]]],
span:organization[Mozilla],
span:title[Fundamentals of Web XR]
span:uri[https://developer.mozilla.org/en-US/docs/Web/API/WebXR_Device_API/Fundamentals]
````

**Comment by opoudjis** [comment]
The formattedref rendering of the foregoing spans, naively, is

`"<bibitem id="mozilla">
<formattedref format="application/x-isodoc+xml">Mozilla, Fundamentals of Web XR<link target="https://developer.mozilla.org/en-US/docs/Web/API/WebXR_Device_API/Fundamentals"/></formattedref>`

That is just grabbing the spans as they were, and the formattedref is not being overwritten with the new formatted reference based on the spans markup. In fact, only the title is being recognised. The URI appears not to be recognised because  URIs inside of the span are still being converted into `link` by Metanorma, with no textual content. (This is a recurring problem.) 

And more damning, the formattedref present with the span markup is not being overwritten. It needs to, if spans have been extracted.

Finally, the template for rendering standards is:

````
"{{ title }}."
````

There is NO provision for either author or URL in IEEE standards rendering currently, by contrast with books:

````
"{{creatornames}}, <em>{{title}}</em>, {{ edition }}, {{place}}: {{publisher}}, {{date}}, {{ labels['viewed'] }}_{{date_accessed}}, {{ uri }} ."
````

19.2. of the style manual explicitly says that standards consist of only designation (which is inserted separately), and title. There is NO provision for creators, or URIs. So either I violate 19.2, or this is not a standard, but e.g. a book (which is what the example in 19.4.6 is).

**Comment by opoudjis** [comment]
So two things for me to do:

* span:uri[...] and span:docidentifier[...] containing URIs need not to have the URIs processed as URIs by Metanorma, and be converted into `<link>`.
* If there are span:[] in formatteduri, the formattedref containing the spans needs to be removed, so that it can be overwritten by a formattedref generated out of the spans-generated bibitem fields.

**Comment by ronaldtse** [comment]
> 19.2. of the style manual explicitly says that standards consist of only designation (which is inserted separately), and title. There is NO provision for creators, or URIs. So either I violate 19.2, or this is not a standard, but e.g. a book (which is what the example in 19.4.6 is).

I suspect that there is some misunderstanding here -- "Fundamentals of Web XR" is clearly NOT a standard.

**Comment by opoudjis** [comment]
Yes, and therefore it needs to be signalled as such, with doctype:[]. The default for references in Metanorma is standard, and now that we are using relaton-render, the distinction matters.

**Comment by opoudjis** [comment]
The missing space between the tile and the URI is because there is a carriage return right after the span, with is marked up as `<span>...</span>`, and trailing carriage return is being deleted at the end of line after `>` before outputting XML to disk; this has been the case for years.

Overriding that locally within formatted ref by inserting space between end-tag and carriage return: within formattedref, we assume mixed text and markup.

**Comment by opoudjis** [comment]
In IEEE, I am going to restrict Title-only rendering to standards; techreport is going to be rendered like books, with URIs and authoring organisations.

**Comment by opoudjis** [comment]
Markup:

````
* [[[mozilla,Mozilla Web XR]]],
span:organization[Mozilla],
span:title[Fundamentals of Web XR]
span:uri[https://developer.mozilla.org/en-US/docs/Web/API/WebXR_Device_API/Fundamentals]

* [[[mozilla1,Mozilla Web XR]]],
span:organization[Mozilla],
span:title[Fundamentals of Web XR]
span:uri[https://developer.mozilla.org/en-US/docs/Web/API/WebXR_Device_API/Fundamentals]
span:type[techreport]
````

Output:

<img width="862" alt="Στιγμιότυπο 2022-11-01, 02 00 34" src="https://user-images.githubusercontent.com/1911039/199039776-7834379a-ec0f-4005-8248-3f8441e2c0e9.png">

**Comment by Intelligent2013** [comment]
@opoudjis thank you!
